web/src/components/PeersStrengthView.res

```diff
@@ -6,6 +6,30 @@
 
 open Prelude
 
+module ConnectionDiagram = {
+  type t = {a1: string, a2: string, s: int}
+  @react.component @module("./connection_diagram")
+  external make: (~data: array<t>) => React.element = "default"
+
+  let adapt = (xs: list<SearchTypes.author_peer>) =>
+    xs
+    ->Belt.List.reduce(list{}, (acc, item) => {
+      let match_reversed = xs->Belt.List.keep(e => e.author == item.peer && e.peer == item.author)
+      let nitem = switch match_reversed->Belt.List.get(0) {
+      | None => {a1: item.author, a2: item.peer, s: item.strength->Int32.to_int}
+      | Some(match) => {
+          a1: item.author,
+          a2: item.peer,
+          s: item.strength->Int32.to_int + match.strength->Int32.to_int,
+        }
+      }
+      acc->Belt.List.has(nitem, (a, b) => a.a1 == b.a1 && a.a2 == b.a2)
+        ? acc
+        : Belt.List.add(acc, nitem)
```

Comment:
Not entirely sure, but perhaps we could skip another list traversal by returning the accumulator in the match:
```suggestion
      | None => acc
      | Some(match) => acc->Belt.List.add({
          a1: item.author,
          a2: item.peer,
          s: item.strength->Int32.to_int + match.strength->Int32.to_int,
        })
      }
```

Reply:
Actually that does not work because we need to remove the previous pair. Perhaps we should be doing the merge with a https://rescript-lang.org/docs/manual/latest/api/belt/hash-set instead. But since we'll have to pass an array to the js, it might be easier to simply merge the value in the js side for now.

Reply:
I had a stab at this, would this work?

```rescript
  let adapt = (peers: list<SearchTypes.author_peer>) => {
    let rec go = (xs: list<SearchTypes.author_peer>, acc: list<t>) =>
      switch xs {
      | list{} => acc
      | list{x, ...rest} => {
          // Look for a reverse peer (and remove it from the remaining list)
          let (otherLink, remaining) =
            rest->Belt.List.partition(e => x.author == e.peer && x.peer == e.author)
          // If there is one, get its strength
          let extraStrength = switch otherLink {
          // here we assume there can only be one other link
          | list{y} => y.strength->Int32.to_int
          | _ => 0
          }
          // Continue with the remaining
          remaining->go(
            acc->Belt.List.add({
              a1: x.author,
              a2: x.peer,
              s: x.strength->Int32.to_int + extraStrength,
            }),
          )
        }
      }
    peers->go(list{})->Belt.List.toArray
  }
```

Reply:
Well ok, even if I think my implementation was accurate, let's move on with yours as it traverses less elements thanks to the Belt.List.partition.

Reply:
Could we merge ?

Reply:
Oh yes the previous code was correct, you'll have to approve the PR to get it merge.